common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -251,15 +251,17 @@ describe 'Problem', ->
               callable()
             done: (callable) ->
               callable()
-        spyOn @problem, 'enableAllButtons'
         @problem.submit()
-        expect(@problem.enableAllButtons).toHaveBeenCalledWith false, true
         if jQuery.active == 0
           deferred.resolve()
         deferred.promise()
+      spyOn(@problem, 'enableSubmitButton').and.callThrough()
+      # Submit button should not be disabled after adding an input
+      $('#input_example_1').val('test').trigger('input')
+      expect(@problem.submitButton).not.toHaveAttr('disabled')
 
       runs.call(self).then(->
-        expect(self.problem.enableAllButtons).toHaveBeenCalledWith true, true
+        expect(self.problem.enableSubmitButton).toHaveBeenCalledWith false, true
```

Comment:
Why is it necessary to use this spy statement instead of checking the disabled state of the submit button?